kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaProxyReconciler.java

```diff
@@ -140,8 +140,13 @@ private static KafkaProxy buildStatus(KafkaProxy primary,
         }
         var now = ZonedDateTime.now(ZoneId.of("Z"));
         // @formatter:off
-        return new KafkaProxyBuilder(primary)
-                .editOrNewStatus()
+        return new KafkaProxyBuilder()
+                .withNewMetadata()
+                    .withName(ResourcesUtil.name(primary))
+                    .withNamespace(ResourcesUtil.namespace(primary))
+                    .withUid(ResourcesUtil.uid(primary))
+                .endMetadata()
+                .withNewStatus()
```

Comment:
Could this be simplified along the lines of 
```suggestion
        return new KafkaProxyBuilder()
               .withNewMetadataLike(primary.getMetadata())
                .endMetadata()
                .editOrNewStatusLike(primary.getStatus())
```

Modulo formatting...

Reply:
No, because `withNewMetadataLike(primary.getMetadata())` will include all the other metadata in the primary, which we explicitly don't want to copy. We don't want to be "field manager" for all the labels, for example, because that would prevent users using `kubectl` to set those labels. Other fields are really owned by the apiserver, like the `resourceVersion` (TBH I don't know exactly _how_ the apiserver behaves if you include those in a SSA patch, but that doesn't change that we shouldn't be trying to set them in a patch).

Reply:
Ok, thats cool. 

I might have a poke at adding a method on ResourceUtil to produce a co-ordinates metadata based on a resource.

Reply:
https://github.com/tombentley/kproxy/pull/20

Reply:
> I might have a poke at adding a method on ResourceUtil to produce a co-ordinates metadata based on a resource.

Reply:
As in, I'm about to push something along the lines I mentioned this morning.